wp_api_integration_tests/tests/test_taxonomies_immut.rs

```diff
@@ -0,0 +1,203 @@
+    SparseTaxonomyTypeDetailsFieldWithViewContext, TaxonomyListParams, TaxonomyType,
+};
+use wp_api_integration_tests::{api_client, AssertResponse};
+
+#[tokio::test]
+#[apply(list_cases)]
+#[parallel]
+async fn list_taxonomies_with_edit_context(#[case] params: TaxonomyListParams) {
+    let response = api_client()
+        .taxonomies()
+        .list_with_edit_context(&params)
+        .await
+        .assert_response()
+        .data;
+    assert_eq!(
+        response
+            .taxonomy_types
+            .get(&TaxonomyType::Category)
+            .expect("Our local WordPress test site has `category` taxonomy type")
+            .name,
+        "Categories"
+    );
+}
+
+#[tokio::test]
+#[apply(list_cases)]
+#[parallel]
+async fn list_taxonomies_with_embed_context(#[case] params: TaxonomyListParams) {
+    let response = api_client()
+        .taxonomies()
+        .list_with_embed_context(&params)
+        .await
+        .assert_response()
+        .data;
+    assert_eq!(
+        response
+            .taxonomy_types
+            .get(&TaxonomyType::Category)
+            .expect("Our local WordPress test site has `category` taxonomy type")
+            .name,
+        "Categories"
+    );
+}
+
+#[tokio::test]
+#[apply(list_cases)]
+#[parallel]
+async fn list_taxonomies_with_view_context(#[case] params: TaxonomyListParams) {
+    let response = api_client()
+        .taxonomies()
+        .list_with_view_context(&params)
+        .await
+        .assert_response()
+        .data;
+    assert_eq!(
+        response
+            .taxonomy_types
+            .get(&TaxonomyType::Category)
+            .expect("Our local WordPress test site has `category` taxonomy type")
+            .name,
+        "Categories"
+    );
+}
+
+#[tokio::test]
+#[apply(retrieve_cases)]
+#[parallel]
+async fn retrieve_taxonomies_with_edit_context(#[case] taxonomy_type: TaxonomyType) {
+    api_client()
+        .taxonomies()
+        .retrieve_with_edit_context(&taxonomy_type)
+        .await
+        .assert_response();
+}
+
+#[tokio::test]
+#[apply(retrieve_cases)]
+#[parallel]
+async fn retrieve_taxonomies_with_embed_context(#[case] taxonomy_type: TaxonomyType) {
+    api_client()
+        .taxonomies()
+        .retrieve_with_embed_context(&taxonomy_type)
+        .await
+        .assert_response();
+}
+
+#[tokio::test]
+#[apply(retrieve_cases)]
+#[parallel]
+async fn retrieve_taxonomies_with_view_context(#[case] taxonomy_type: TaxonomyType) {
+    api_client()
+        .taxonomies()
+        .retrieve_with_view_context(&taxonomy_type)
+        .await
+        .assert_response();
+}
+
+#[template]
+#[rstest]
+#[case::default(TaxonomyListParams::default())]
+#[case::post(TaxonomyListParams { post_type: Some(PostType::Post) })]
+// TODO: These post types return:
+// ```
+// {"code":"rest_cannot_view","message":"Sorry, you are not allowed to manage terms in this taxonomy.","data":{"status":403}}
```

Comment:
I used a jurrasic ninja site (with pre-populated content) for testing. And I only see this "Sorry, you are not allowed to manage terms in this taxonomy." error when using `context=edit` and a type that does not have taxonomies.

For example, `/wp/v2/taxonomies` in my test site returns "Pattern Categories", which has `wp_block` in its `types`, and `/wp/v2/taxonomies?type=wp_block&context=edit` returns a valid response for me. But there is no `wp_template` results in `/wp/v2/taxonomies`, and `wp/v2/taxonomies?type=wp_template&context=edit` returns the "Sorry, you are not allowed to manage terms in this taxonomy" error.

Reply:
@crazytonyli Thanks a lot for looking into this. Your explanation clarified how it works and I found that we have `post`, `nav_menu_item` & `wp_block` post types in our test site, so I've added tests for them in d3736919fe7523c171b10e5b731bd6fff8ba111c.